packages/core/test/integration/query-interface/add-show-remove-index.test.ts

```diff
@@ -0,0 +1,304 @@
+            const indexCheckAfterRemoval = indexesAfterRemoval.find(index => index.name === 'user_username');
+
+            expect(indexCheckAfterRemoval).to.be.undefined;
+          } else {
+            expect.fail('This should have failed');
+          }
+        } catch (error) {
+          expect(error).to.be.instanceOf(DatabaseError);
+          assert(error instanceof DatabaseError);
+          expect(error.message).to.match(/\s|^Cannot use duplicate column names in index. Column name 'username' listed more than once.$/);
+        }
+      });
+
+      it('throws an error with missing column names in included columns', async () => {
+        try {
+          await queryInterface.addIndex('users', { name: 'user_username', fields: ['username'], include: ['first_name', 'last_name', 'email'], unique: true });
+          expect.fail('This should have failed');
+        } catch (error) {
+          expect(error).to.be.instanceOf(DatabaseError);
+          assert(error instanceof DatabaseError);
+          expect(error.message).to.match(/\s|^Column name 'email' does not exist in the target table or view.$/);
+        }
+      });
+
+      it('throws an error with invalid column type with included columns', async () => {
+        await queryInterface.addColumn('users', 'bio', DataTypes.TEXT);
+
+        try {
+          await queryInterface.addIndex('users', { name: 'user_username', fields: ['bio'], include: ['first_name', 'last_name'], unique: true });
+          if (dialect === 'postgres') {
+            const indexes = await sequelize.queryInterface.showIndexes('users');
+            const indexCheck = indexes.find(index => index.name === 'user_username');
+
+            expect(indexCheck).to.not.be.undefined;
+            assert(indexCheck);
+            expect(indexCheck.tableName).to.equal('users');
+            expect(indexCheck.name).to.equal('user_username');
+            expect(indexCheck.unique).to.equal(true);
+            expect(indexCheck.fields).to.have.length(1);
+            expect(indexCheck.fields[0].name).to.equal('bio');
+            expect(indexCheck.includes).to.have.length(2);
+            expect(indexCheck.includes).to.include('first_name');
+            expect(indexCheck.includes).to.include('last_name');
+
+            await queryInterface.removeIndex('users', 'user_username');
+            const indexesAfterRemoval = await sequelize.queryInterface.showIndexes('users');
+            const indexCheckAfterRemoval = indexesAfterRemoval.find(index => index.name === 'user_username');
+
+            expect(indexCheckAfterRemoval).to.be.undefined;
+          } else {
+            expect.fail('This should have failed');
+          }
+        } catch (error) {
+          expect(error).to.be.instanceOf(DatabaseError);
+          assert(error instanceof DatabaseError);
+          expect(error.message).to.match(/\s|^Column 'username' in table 'users' is of a type that is invalid for use as a key column in an index.$/);
```

Comment:
this just tests database limits instead of the ORM, I don't think this test is relevant for us

Reply:
Updated now.